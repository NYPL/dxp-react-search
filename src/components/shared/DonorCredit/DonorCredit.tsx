import * as React from "react";
import {
  Heading,
  Box,
  HorizontalRule,
  Flex,
} from "@nypl/design-system-react-components";
import TextFormatted from "./../TextFormatted";

interface DonorCreditProps {
  id: string;
  showBorder: boolean;
  heading?: string;
  description: string;
}

export default function DonorCredit({
  id,
  showBorder,
  heading,
  description,
}: DonorCreditProps) {
  return (
    <Flex id={`donor-credit-${id}`} marginBottom="xl" flexDirection="column">
      {showBorder && <HorizontalRule marginBottom="xl" marginTop="0" />}
      {heading && <Heading level="three" text={heading} marginBottom="l" />}
      <Box
        maxW={{ lg: "850px" }}
        margin="0 auto"
        px={{ sm: "s", md: "l", lg: "0" }}
        sx={{
          "& p": {
            fontStyle: "italic",
          },
        }}
      >
        <TextFormatted html={description} />
      </Box>
    </Flex>
  );
}
