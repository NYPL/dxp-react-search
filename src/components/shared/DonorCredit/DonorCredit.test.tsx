import * as React from "react";
import { render, screen } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import "@testing-library/jest-dom/extend-expect";
expect.extend(toHaveNoViolations);
// Component
import DonorCredit from "./DonorCredit";
import renderer from "react-test-renderer";

describe("DonorCredit tests", () => {
  it("should pass axe accessibility test", async () => {
    const { container } = render(
      <DonorCredit
        showBorder={true}
        description="Test Description"
        id="Test ID"
        heading="Test Heading"
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("should render the UI snapshot correctly", () => {
    const defaultDonorCredit = renderer
      .create(
        <DonorCredit
          showBorder={true}
          description="Test Description"
          id="Test ID"
          heading="Test Heading"
        />
      )
      .toJSON();

    const donorCreditWithoutBorder = renderer
      .create(
        <DonorCredit
          showBorder={false}
          description="Test Description"
          id="Test ID"
          heading="Test Heading"
        />
      )
      .toJSON();

    const donorCreditWithoutHeading = renderer
      .create(
        <DonorCredit
          showBorder={false}
          description="Test Description"
          id="Test ID"
        />
      )
      .toJSON();

    expect(defaultDonorCredit).toMatchSnapshot();
    expect(donorCreditWithoutBorder).toMatchSnapshot();
    expect(donorCreditWithoutHeading).toMatchSnapshot();
  });

  it("should render no heading element when heading is empty or all spaces", () => {
    const { rerender } = render(
      <DonorCredit
        showBorder={true}
        description="Test Description"
        id="Test ID"
        heading="   "
      />
    );

    expect(screen.queryByRole("heading")).not.toBeInTheDocument();

    rerender(
      <DonorCredit
        showBorder={true}
        description="Test Description"
        id="Test ID"
        heading=""
      />
    );

    expect(screen.queryByRole("heading")).not.toBeInTheDocument();
  });
});
