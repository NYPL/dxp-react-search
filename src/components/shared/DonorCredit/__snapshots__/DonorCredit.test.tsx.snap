// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonorCredit tests should render the UI snapshot correctly 1`] = `
<div
  className="css-8hxh7n"
  id="donor-credit-Test ID"
>
  <hr
    className="css-1auw00o"
  />
  <h3
    className="chakra-heading css-1e3lsoz"
  >
    Test Heading
  </h3>
  <div
    className="css-144igvy"
  >
    <div
      className="css-y1avno"
      dangerouslySetInnerHTML={
        {
          "__html": "Test Description",
        }
      }
    />
  </div>
</div>
`;

exports[`DonorCredit tests should render the UI snapshot correctly 2`] = `
<div
  className="css-8hxh7n"
  id="donor-credit-Test ID"
>
  <h3
    className="chakra-heading css-1e3lsoz"
  >
    Test Heading
  </h3>
  <div
    className="css-144igvy"
  >
    <div
      className="css-y1avno"
      dangerouslySetInnerHTML={
        {
          "__html": "Test Description",
        }
      }
    />
  </div>
</div>
`;

exports[`DonorCredit tests should render the UI snapshot correctly 3`] = `
<div
  className="css-8hxh7n"
  id="donor-credit-Test ID"
>
  <div
    className="css-144igvy"
  >
    <div
      className="css-y1avno"
      dangerouslySetInnerHTML={
        {
          "__html": "Test Description",
        }
      }
    />
  </div>
</div>
`;
